components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -3894,6 +3894,12 @@ void RewardsServiceImpl::ClearAllNotifications() {
 
 void RewardsServiceImpl::CompleteReset(SuccessCallback callback) {
   notification_service_->DeleteAllNotifications(true);
+
+  // Close any open files before deleting them below (required on
+  // Windows)
+  file_task_runner_->DeleteSoon(FROM_HERE, rewards_database_.release());
```

Comment:
Looks like `OnCompleteReset` calls `StopLedger` which does this delete of the rewards database. Does `CompleteReset` need to call `StopLedger` before deleting all of the files? Like that would be the last thing that it does instead of the first?

Reply:
I'm not opposed to something like that, but `StopLedger` is also asynchronous. I'm a little leery about changing this too deeply, as there could be some unintended side effects (just ran into one with the staging server on Android yesterday in fact).

Reply:
In the ideal world, it seems like the `OnCompleteReset` callback should do nothing but call observers and run its success callback. `CompleteReset` should shutdown ledger and then (when that completes) close and delete files. It's definitely a little more complicated than the current solution, but also feels a bit more "correct". What do you think @NejcZdovc?

Reply:
@emerick sure, adjust PR and let's see how it will work